### DISCUSSION TITLE
Unable to see real client IP address for DoT/DoH

### DISCUSSION THREAD

**Discussion by prashantkamdar** [AdguardTeam/AdGuardHome#4652] [Q&A] [answered]
Apologies 

This is not a bug or a feature request, but a help request:

I am using dockerised adguard with nginx. Everything is working fine including DoT/DoH.
The problem I am facing is that the real IP address is not showing up when doing DoT queries in the query log.
The IP that shows up is the docker's IP on the host server - 172.17.0.1

![image](https://user-images.githubusercontent.com/2802365/172865288-d158c29d-49f1-4d2f-92de-a7aff2e8de83.png)

My nginx is built with `--with-http_realip_module` as prescribed in the adguard's docs.

I have tried the following:
```
server
{
  listen 443 ssl;
  access_log /var/log/nginx/adguard.access.log;
  error_log /var/log/nginx/adguard.error.log;
  ssl_certificate domain.cert.pem;
  ssl_certificate_key private.key.pem;
  server_name adguard.7thking.xyz;

  location /
  {
    proxy_pass http://127.0.0.1:1080/;
    proxy_redirect / /;
    proxy_cookie_path / /;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }

  location /dns-query
  {
    proxy_pass http://127.0.0.1:1080;
    proxy_bind 127.0.0.1;
    proxy_set_header        X-Real-IP          $remote_addr;
    proxy_set_header        X-Forwarded-For    $proxy_add_x_forwarded_for;
  }
}
```

But I am not getting anywhere.

I have the following in my AdGuardHome.yaml for trusted proxies:
```
  trusted_proxies:
  - 127.0.0.0/8
  - ::1/128
   ```

Can someone please guide how to get the IP for DoT/DoH with nginx and dockered adguardhome?

**Comment by ainar-g** [comment, marked as answer]
If the queries look like they're coming from `172.17.0.1`, consider adding that address to the `trusted_proxies`.

**Reply by prashantkamdar** [reply]
thank you!
that worked!

**Comment by quyleanh** [comment]
@prashantkamdar could you please show your full docker/docker-compose config?

I have a problem described [here](https://github.com/AdguardTeam/AdGuardHome/discussions/4656) and would like to refer your config. Thank you.

**Reply by prashantkamdar** [reply]
I just did docker run 

```
docker run --name adguardhome\ 
    --restart unless-stopped\ 
    -v ~/adguard/work:/opt/adguardhome/work\ 
    -v ~/adguard/conf:/opt/adguardhome/conf\ 
    -p 53:53/tcp -p 53:53/udp\ 
    -p 1080:80/tcp -p 10443:443/tcp -p 10443:443/udp -p 3000:3000/tcp\ 
    -p 853:853/tcp\ 
    -p 784:784/udp -p 853:853/udp -p 8853:8853/udp\ 
    -p 5443:5443/tcp -p 5443:5443/udp\ 
    -d adguard/adguardhome
```

Once done, I had to enable ipv6 for docker using this tut: https://medium.com/@skleeschulte/how-to-enable-ipv6-for-docker-containers-on-ubuntu-18-04-c68394a219a2

**Reply by quyleanh** [reply]
Thank you for your response.

So you can access to dashboard by enter normal domain (without :1080 or :10443 postfix) right?
Are you using https for dashboard site?

**Reply by prashantkamdar** [reply]
I have nginx reverse proxy on the same server 
and yes, i access the dashboard on https

this is my nginx conf:

```
server
{
  listen 443 ssl;
  access_log /var/log/nginx/adguard.access.log;
  error_log /var/log/nginx/adguard.error.log;
  ssl_certificate domain.cert.pem;
  ssl_certificate_key private.key.pem;
  server_name DOMAIN_NAME.COM;

  location /
  {
    proxy_pass http://127.0.0.1:1080/;
    proxy_redirect / /;
    proxy_cookie_path / /;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }

  location /dns-query
  {
    proxy_pass http://127.0.0.1:1080;
    proxy_bind 127.0.0.1;
    proxy_set_header        X-Real-IP          $remote_addr;
    proxy_set_header        X-Forwarded-For    $proxy_add_x_forwarded_for;
  }
}
```

**Reply by quyleanh** [reply]
How can. I did almost the same as your.

Could you please take a look at [my discussion](https://github.com/AdguardTeam/AdGuardHome/discussions/4656)?

**Reply by prashantkamdar** [reply]
Follow the reco given - https://github.com/AdguardTeam/AdGuardHome/discussions/4656#discussioncomment-2923170

**Reply by quyleanh** [reply]
I did but it didn't help :'(
